lib/grandpa/network.go

```diff
@@ -222,8 +226,14 @@ func decodeMessage(cm *network.ConsensusMessage) (m GrandpaMessage, err error) {
 		m = &val
 	case CommitMessage:
 		m = &val
-	case NeighbourMessage:
-		m = &val
+
+	case VersionedNeighbourPacket:
+		switch NeighbourMessage := val.Value().(type) {
+		case NeighbourPacketV1:
+			m = &NeighbourMessage
+		default:
+			return nil, ErrInvalidMessageType
```

Comment:
Shouldn't we panic in this case? That looks like a programming error right?

Reply:
Not really, sometimes is just an unsupported neighbor packet, and since this code deals with outside data inserting a panic there could be an attack vector by allowing peers to send us wrong neighbor messages that we don't support and then panicking us